app/src/main/java/com/github/factotum_sdp/factotum/data/LoginRepository.kt

```diff
@@ -0,0 +1,46 @@
+package com.github.factotum_sdp.factotum.data
+
+import com.github.factotum_sdp.factotum.data.model.LoggedInUser
+
+/**
+ * Class that requests authentication and user information from the remote data source and
+ * maintains an in-memory cache of login status and user credentials information.
+ */
+
+class LoginRepository(val dataSource: LoginDataSource) {
+
+    // in-memory cache of the loggedInUser object
+    var user: LoggedInUser? = null
+        private set
+
+    val isLoggedIn: Boolean
+        get() = user != null
+
+    init {
+        // If user credentials will be cached in local storage, it is recommended it be encrypted
+        // @see https://developer.android.com/training/articles/keystore
+        user = null
+    }
+
+    fun logout() {
+        user = null
+        dataSource.logout()
+    }
+
+    fun login(username: String, password: String): Result<LoggedInUser> {
```

Comment:
Nitpicking here, but it makes more sense in my opinion to have the login function before the logout function.